spdx-utils/src/main/kotlin/SpdxSimpleLicenseMapping.kt

```diff
@@ -47,8 +47,7 @@ object SpdxSimpleLicenseMapping {
      */
     private val deprecatedLicenseIds by lazy {
         val resource = javaClass.getResource("/deprecated-license-mapping.yml")
-        val map = yamlMapper.readValue<Map<String, SpdxLicense>>(resource)
-        map.mapValues { (_, v) -> v.toExpression() }
+        yamlMapper.readValue<Map<String, SpdxSingleLicenseExpression>>(resource)
```

Comment:
Shouldn't that be `SpdxLicenseIdExpression`? That's the type it had before.

Reply:
You're basically right, but unfortunately this does not work as the default constructor of `SpdxLicenseIdExpression` sets `orLaterVersion` to `false`, which means it fails to deserialize e.g. an `id` of "AGPL-1.0-or-later", which would require `orLaterVersion` to be set to `true`. The direct parent class `SpdxSimpleExpression` also cannot be used as it has no constructor. So I'm using the grand-parent class `SpdxSingleLicenseExpression`, which also is a preparation towards merging the mapping for licenses and excpetions, which I was planning to do.